va/validation-authority.go

```diff
@@ -523,7 +499,11 @@ func (va *ValidationAuthorityImpl) validateDNS(identifier core.AcmeIdentifier, i
 		}
 		return challenge, err
 	}
-	encodedSignature := core.B64enc(challenge.Validation.Signatures[0].Signature)
+
+	// Compute the digest of the authorized keys file
+	h := sha256.New()
+	h.Write([]byte(challenge.AuthorizedKeys))
+	authorizedKeysDigest := hex.EncodeToString(h.Sum(nil))
```

Comment:
I like the fact that a hash is used here, this will simplify the dns validation implementation by not requiring to store long txt records.